lib/providers/map_data.dart

```diff
@@ -24,11 +24,23 @@ class MapData extends ChangeNotifier {
   LatLng _start;
   LatLng _end;
   String _mode;
+  String _duration;
+  String _distance;
 
   MapData() {
     _mode = "driving";
   }
 
+  void changeDuration(duration) {
+    _duration = duration;
+    notifyListeners();
+  }
+
+  void changeDistance(distance) {
+    _distance = distance;
+    notifyListeners();
+  }
+
```

Comment:
Redundant, as the shared itinerary object already contains distance and duration